src/domain/event/eventCalendarButton/EventCalendarButton.tsx

```diff
@@ -0,0 +1,38 @@
+import { ButtonProps, ButtonVariant, IconCalendar } from 'hds-react';
+import { useTranslation } from 'next-i18next';
+import { FC } from 'react';
+
+import Button from '../../../common/components/button/Button';
+import { useNotificationsContext } from '../../../common/components/notificationsContext/hooks/useNotificationsContext';
+import useLocale from '../../../hooks/useLocale';
+import { Event } from '../types';
+import { downloadEventIcsFile } from '../utils';
+
+export type EventCalendarButtonProps = {
+  event: Event;
+} & Omit<ButtonProps, 'children' | 'onClick'>;
+
+const EventCalendarButton: FC<EventCalendarButtonProps> = ({
+  event,
+  variant = 'secondary',
+  ...rest
+}) => {
+  const locale = useLocale();
+  const { t } = useTranslation('common');
+  const { addNotification } = useNotificationsContext();
+
+  return (
+    <Button
+      {...rest}
+      iconLeft={<IconCalendar aria-hidden />}
+      onClick={() =>
+        downloadEventIcsFile({ addNotification, event, locale, t })
```

Comment:
Maybe the download function could be passed as a prop like variant to allow using a different function too?

Reply:
Added support for custom onClick function.